### PR TITLE
Issue #104 - Add integration tests for Che addon

### PR DESCRIPTION
**Pull request by agajdosi**
Migrated original PR from minishift repo to this repository. Tests works now.
I have tried to clean it up a little bit, but it is possible it was not enough, let me know your opinion in review.

The feature add package Che, which drives most of the Che feature file. In the feature M is started and several workspaces are started, some example project is deployed on them and they are deleted, at the end M is deleted.

What I do not like that gherkin steps are little bit cryptic in this PR, but we might just live with it. Tried to make visible as much as was possible, more visibility would need considerable refactor. 

To test this PR run:
```
make clean
make vendor
make integration ADDON=che
```

Let me know your ideas, @minishift/minishift-dev.

### CODE REVIEW

File: test/integration/features/addon-che.feature [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    When executing "minishift addons list" succeeds
+    Then stdout should contain "che"
+  
+  Scenario: User starts Minishift
+    Given Minishift has state "Does Not Exist"
+    When executing "minishift start --memory=5GB" succeeds
+    Then Minishift should have state "Running"
+
+  Scenario: Applying Che add-on
+    #When applying openshift token succeeds
```
Comment (budhrg) [inline test/integration/features/addon-che.feature]: Unnecessary comments.

File: add-ons/che/che.addon [hunk 2]
```diff
@@ -4,6 +4,7 @@
 # Required-Vars: NAMESPACE, CHE_DOCKER_IMAGE, OPENSHIFT_TOKEN, GITHUB_CLIENT_ID, GITHUB_CLIENT_SECRET
 # Var-Defaults: NAMESPACE=mini-che, CHE_DOCKER_IMAGE=eclipse/che-server:latest, OPENSHIFT_TOKEN=NULL, GITHUB_CLIENT_ID=changeme, GITHUB_CLIENT_SECRET=changeme
 # OpenShift-Version: >=3.5.0
+# JUST TO TRIGGER THE TESTS
```
Comment (budhrg) [inline add-ons/che/che.addon]: Unnecessary comment.

File: test/integration/features/addon-che.feature [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    When stopping a workspace succeeds
+    Then workspace should have state "STOPPED"
+
+    When workspace is removed
+    Then workspace removal should be successful
+
+    Examples:
+    | stack                 | sample                                                                   |
+    | Eclipse Vert.x        | https://github.com/openshiftio-vertx-boosters/vertx-http-booster         |
```
Comment (budhrg) [inline test/integration/features/addon-che.feature]: We should only run one stack. 

May be Java one. WDYT @LalatenduMohanty ?

Comment (budhrg) [reply test/integration/features/addon-che.feature]: @agajdosi @LalatenduMohanty , let it be like this in upstream. As, it runs as part of nightly test only.

For downstream, we need to decide what to test.

**Comment by agajdosi** [comment]
@budhrg Removed the extra comments.

**Pull request merged**